system/Database/BaseConnection.php

```diff
@@ -373,8 +373,15 @@ public function initialize()
 
 		$this->connectTime = microtime(true);
 
-		// Connect to the database and set the connection ID
-		$this->connID = $this->connect($this->pConnect);
+		try 
+		{
+			// Connect to the database and set the connection ID
+			$this->connID = $this->connect($this->pConnect);
+		} 
+		catch (\Throwable $e) 
+		{
+			log_message('error', 'connect database error: ' . $e->getMessage());
```

Comment:
Slight change to the wording, here and below:
`log_message('error', 'Error connecting to the database: ' . $e->getMessage());`

Reply:
Okay thank you. Sorry, English is not my mother tongue, so some descriptions are wrong, I translated through Google.